### ISSUE TITLE
Cannot add @angular/pwa to my angular project

### ISSUE THREAD

**Issue by sir-babagee** [angular/angular-cli#25399]
### Command

add

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

I'm trying to add @angular/pwa to my angular project but I keep running into this error.

`× Unable to load package information from registry: The "timeout" argument must be of type number. Received type string ('6000000')`

I've been looking and I can't seem to find anywhere where this error occurred. I would appreciate if a solution to this can be provided.

### Minimal Reproduction

Can't really say what caused the issue or how to reproduce as this project is pretty large and has been in the works for a while

### Your Environment

```text
Angular CLI: 15.2.7
Node: 18.16.0
Package Manager: npm 9.7.1
OS: win32 x64

Angular: 16.1.1
... core

Package                             Version
-------------------------------------------------------------
@angular-devkit/architect           0.1502.7
@angular-devkit/build-angular       15.2.7
@angular-devkit/core                16.1.0
@angular-devkit/schematics          16.1.0
@angular/animations                 15.2.8
@angular/cdk                        15.2.8
@angular/cli                        15.2.7
@angular/common                     15.2.8
@angular/compiler                   15.2.8
@angular/compiler-cli               15.2.8
@angular/fire                       7.5.0
@angular/forms                      15.2.8
@angular/material                   15.2.8
@angular/platform-browser           15.2.8
@angular/platform-browser-dynamic   15.2.8
@angular/router                     15.2.8
@schematics/angular                 16.1.0
rxjs                                7.8.1
typescript                          4.8.4
```

**Comment by clydin** [comment]
Is there a `timeout` field in an `.npmrc` file used by the project? This could be an `.npmrc` within the project directory or a global file.

Also, it appears that the project is in a partially updated state with some framework dependencies on v16 of Angular and most on v15. This is not a supported configuration and may lead to errors or unexpected behavior.
The `ng update` command can be used to update the project with additional guidance found here: https://update.angular.io/

**Comment by sir-babagee** [comment]
I found a work around for this. The issue (what I think) has to do with the fact that this project has issues with installing dependencies, the issue being that each package that needs to be installed has to have --legacy-peer-deps or --force flag attached to it before it installs successfully. So probably what was happening was when I used the `ng add` command, it didn't seem to be able to become compatible with the project and it stops. So what I did was to install the @angular/pwa like so: `npm install @angular/pwa --legacy-peer-deps` for it to be installed, then added the necessary files for it to be used.
